packages/driver/src/cypress/keyboard.js

```diff
@@ -575,9 +580,11 @@ const $Keyboard = {
           return
         }
 
+        const isFirstSymbol = !valueLength || (selectionLength === valueLength)
+
         //# only type '.' and '-' if it is the first symbol and there already is a value, or if
         //# '.' or '-' are appended to a digit. If not, value cannot be set.
```

Comment:
I think there may be some missing logic still to match the spec - notably allowing `e`, `E`, and `+`. Also - I'd like a link to the spec in the comments if possible because otherwise I'm just like - where did this logic definition even come from? 

- value allows floating point number https://www.w3.org/TR/2010/WD-html-markup-20101019/input.number.html#input.number.attrs.value
- floating point number definition https://www.w3.org/TR/2010/WD-html-markup-20101019/datatypes.html#common.data.float